### PR TITLE
feat(vendure): Included address based tax zone strategy

### LINKED ISSUE #2914: Include Address based tax zone strategy in core
**Is your feature request related to a problem? Please describe.**
We have implemented a custom TaxZoneStrategy that determines the taxzone based on the orders shipping (or billing) address. Most of our clients tell us they already expect Vendure to work like this, because 'Why else doe we have tax zones'.

We can easily implement this in each project, but since it's such a common use case, it might make sense to have Vendure core export this strategy?

**Describe the solution you'd like**
1. As a developer I would like to be able to set the AddressBasedTaxZoneStrategy in my project:
```ts
import { AddressBasedTaxZoneStrategy } from '@vendure/core';

// vendure-config.ts
  taxOptions: {
    taxZoneStrategy: new ShippingBasedTaxZoneStrategy(),
  },
```

**Describe alternatives you've considered**
Package it in a plugin.

**Additional context**
The strategy we use:
```ts
import {
  Channel,
  Logger,
  Order,
  RequestContext,
  Zone
} from '@vendure/core';
import { TaxZoneStrategy } from '@vendure/core/dist/config/tax/tax-zone-strategy';

const loggerCtx = 'AddressBasedTaxZoneStrategy';

export class AddressBasedTaxZoneStrategy implements TaxZoneStrategy {

  determineTaxZone(
    ctx: RequestContext,
    zones: Zone[],
    channel: Channel,
    order?: Order
  ): Zone {
    const countryCode = order?.billingAddress?.countryCode ?? order?.shippingAddress?.countryCode;
    if (order && countryCode) {
      const zone = zones.find((zone) =>
        zone.members?.find((member) => member.code === countryCode)
      );
      if (zone) {
        return zone;
      }
      Logger.warn(
        `No tax zone found for country ${countryCode}. Returning default ${channel.defaultTaxZone.name} for order ${order.code}`,
        loggerCtx
      );
    }
    return channel.defaultTaxZone;
  }
}
```

### PR DESCRIPTION
**Pull request by martijnvdbrug**
# Description

Implemented address based tax zone strategy that determines tax zone based on the billing or shipping address of the order. Closes #2914

# Breaking changes

None.

# Checklist

📌 Always:
- [x] I have set a clear title
- [x] My PR is small and contains a single feature
- [x] I have [checked my own PR](## "Fix typo's and remove unused or commented out code")

👍 Most of the time:
- [ ] I have added or updated test cases
- [x] I have updated the README if needed

### CODE REVIEW

File: packages/core/src/config/tax/address-based-tax-zone-strategy.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * :::info
+ *
+ * This is configured via `taxOptions.taxZoneStrategy = new AddressBasedTaxZoneStrategy()` in
+ * your VendureConfig.
+ *
+ * :::
+ *
+ * @docsCategory tax
+ */
```
Comment (martijnvdbrug) [inline packages/core/src/config/tax/address-based-tax-zone-strategy.ts:26]: Is this enough to include it in the docs?

Comment (michaelbromley) [reply packages/core/src/config/tax/address-based-tax-zone-strategy.ts:26]: Yes that's fine

**Comment by martijnvdbrug** [comment]
I will leave the doc generation up to you if that is ok. It generates 52 changed files on my local machine. Not sure why

**Comment by michaelbromley** [comment]
That's fine. Actually could you make this PR against the `minor` branch since it is a new feature?

**Comment by michaelbromley** [comment]
Thanks! I think I'll make this the new default for v3.1. It just makes much more sense.

**Pull request merged**